modules/programs/micro/default.nix

```diff
@@ -0,0 +1,41 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.programs.micro;
+  dir = ".config/micro";
+in {
+  meta.maintainers = [ maintainers.sivizius ];
+
+  ###### interface
+  options = {
+    programs.micro = {
+      enable = lib.mkEnableOption
+        "micro – a modern easy-to-use intuitive terminal-based text editor.";
+      bindings = import ./bindings.nix lib;
+      colorschemes = import ./colorschemes.nix lib;
+      extraSettings = import ./extraSettings.nix lib;
+      plugins = import ./plugins.nix lib;
+      settings = import ./settings.nix lib;
+    };
+  };
+
+  ###### implementation
+  config = lib.mkIf cfg.enable {
+    home = let
+      colorschemes = lib.mapAttrs' (name: configuration:
+        lib.nameValuePair ("${dir}/colorschemes/${name}.micro") ({
+          text = lib.concatStringsSep "" (lib.mapAttrsToList
```

Comment:
Try making this code more understandable. Perhaps by adding a few function such as.
```
mkColorScheme = …;
```